src/transformers/models/deta/modeling_deta.py

```diff
@@ -1929,21 +1939,25 @@ def forward(
                 focal_alpha=self.config.focal_alpha,
                 losses=losses,
                 num_queries=self.config.num_queries,
+                assign_first_stage=self.config.assign_first_stage,
+                assign_second_stage=self.config.assign_second_stage,
             )
             criterion.to(logits.device)
             # Third: compute the losses, based on outputs and labels
             outputs_loss = {}
             outputs_loss["logits"] = logits
             outputs_loss["pred_boxes"] = pred_boxes
+            outputs_loss["init_reference"] = init_reference
             if self.config.auxiliary_loss:
-                intermediate = outputs.intermediate_hidden_states if return_dict else outputs[4]
-                outputs_class = self.class_embed(intermediate)
-                outputs_coord = self.bbox_embed(intermediate).sigmoid()
```

Comment:
Will this change the calculated aux loss? Are outputs_coord and outputs_class calculated in an equivalent way above somewhere?

Reply:
@amyeroberts Yes there are two different class_embed which is from decoder and pure class embed. This is the expected behavior!